pennylane/qnodes/base.py

```diff
@@ -578,9 +578,13 @@ def evaluate(self, args, kwargs):
             self._construct(args, kwargs)
 
         self.device.reset()
-        ret = self.device.execute(
-            self.circuit.operations, self.circuit.observables, self.variable_deps
-        )
+
+        if isinstance(self.device, qml.QubitDevice):
```

Comment:
Shall a comment for deprecation be added? Devices like ``default_gaussian`` would also potentially make use of such a refactor, and so the logic could be moved up to the ``Device`` level.

Reply:
Ideally yes, _but_ I don't think we want users to see the deprecation warning, only developers. So it might be better to put this in the docs when we re-write the 'writing a plugin' page.

Reply:
(If we put a deprecation warning here, it will show up every time a user evaluates a qnode using a plugin that inherits from Device). Maybe when more devices are ported, we put one here?

Reply:
Sounds good! My original idea was also just to add a one-line comment in the code itself as a warning, would that be useful now or should it come later?

Reply:
Oh, a comment would be good! I thought you meant a `warning.warning(DeprecationWarning)` :laughing:

Reply:
Nope, I think you are right that that would be too much pressure for users :smile: